src/viewer/upload/FileUpload.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import { Alert, Button, Form, ProgressBar } from "react-bootstrap"
+import axios from 'axios';
+
+function UploadAlert(message) {
+  var msg = message.message
+  if (msg === "File Uploaded") {
+    return (
+      <Alert variant="success">
+        {msg}
+      </Alert>
+    );
+  } else {
+    return (
+      <Alert variant="danger">
+        {msg}
+      </Alert>
+    );
+  }
+}
+
+const FileUpload = () => {
+  const [file, setFile] = useState('');                        // The file that is selected to be uploaded
+  const [uploadedFile, setUploadedFile] = useState({});        // The file that has been uploaded
+  const [message, setMessage] = useState("");                  // The message to return to the user
+  const [uploadPercentage, setUploadPercentage] = useState(0); // The percentage of the upload
+  const [uploading, setUploading] = useState(false)            // Whether a file is uploading
+
+  const onChange = event => {
+    setFile(event.target.files[0]);
```

Comment:
I might add a quick safety check here to make sure files is not null prior to getting the first element.